core/modules/layout/layout.admin.inc

```diff
@@ -232,6 +232,10 @@ function layout_settings_form($form, &$form_state, Layout $layout) {
   // Get the list of layout options.
   $all_layout_info = layout_get_layout_info();
   foreach ($all_layout_info as $layout_name => $layout_info) {
+    // Don't reveal hidden layouts.
+    if ($layout_info['hidden']) {
+      continue;
+    }
```

Comment:
Hidden layouts templates should be shown if they are enabled. Hidden layouts templates should only be hidden on the settings page for enabling layouts (if they are not currently enabled).